frontend/src/components/widgets/FileUploader/FileUploader.tsx

```diff
@@ -64,62 +74,105 @@ class FileUploader extends React.PureComponent<Props, State> {
     if (prevProps.disabled !== this.props.disabled && this.props.disabled) {
       this.reset()
     }
+
+    const currentMaxSize = this.props.element.get("maxUploadSizeMb")
+    if (prevProps.element.get("maxUploadSizeMb") !== currentMaxSize) {
+      this.setState({ maxSizeBytes: sizeConverter(currentMaxSize, "mb", "b") })
+    }
   }
 
-  private dropHandler = (
-    acceptedFiles: File[],
-    rejectedFiles: File[],
-    event: React.SyntheticEvent<HTMLElement>
+  public reset = (): void => {
+    this.setState({
+      status: FileStatuses.READY,
+      errorMessage: undefined,
+      files: [],
+    })
+  }
+
+  public dropHandler = (
+    acceptedFiles: ExtendedFile[],
+    rejectedFiles: FileRejection[]
   ): void => {
     const { element } = this.props
-    const maxSizeMb = element.get("maxUploadSizeMb")
-
-    if (rejectedFiles.length > 0) {
-      // TODO: Tell user which files *are* allowed.
-      const errorMessage = `${rejectedFiles[0].type} files are not allowed`
-      this.setState({
-        status: "ERROR",
-        errorMessage,
-      })
+    const multipleFiles = element.get("multipleFiles")
 
-      return
+    if (!multipleFiles && this.state.files.length) {
+      // Only one file is allowed. Delete existing file.
+      this.delete(this.state.files[0].id || "")
```

Comment:
This logic can change slightly in an upcoming PR. From product review, determined that the replace functionality should only cause one rerun.